### DISCUSSION TITLE
[FIXED] Voucher applied to any product, not just specific products

### DISCUSSION THREAD

**Discussion by Anubiella** [saleor/saleor#10018] [General] [answered]
Hi everyone,

weird situation here ,

version:
![Configuration - Ale Bike Wear 17-6-2022 17-08-29](https://user-images.githubusercontent.com/31857092/174325563-a37268bd-c62e-45d2-8253-c7d6cf72c86e.png)

So basically we've created a simple voucher of 30% on specific products (43 to be precise), but just once per order, as
you can see:
![Vouchers - Ale Bike Wear 17-6-2022 17-14-28](https://user-images.githubusercontent.com/31857092/174327109-4c201710-53b5-45aa-8ce0-67e9a9c401fe.png)

Now what happen is that when we try to apply the promoCode to another product that is not in the list,
it gives back no error and apply the voucher as if it is correct.

Meanwhile I've applied an hardcode check on the code and the cart products,
but It's necessary to understand this behaviour.

Can you help me with this issue?

Thanks a lot in advance,
god bless

Lorenzo

**Comment by timuric** [comment, marked as answer]
@Anubiella this option had a misleading description, we have renamed it to `Apply only to a single cheapest eligible product` in 3.4. 
Probably you would like to uncheck this box.

**Reply by Anubiella** [reply]
Got it, thanks a lot!

I'll check the discounts and related checkboxes.

Thanks again,
have a nice day
god bless
